src/states/node.rs

```diff
@@ -881,6 +886,12 @@ impl Node {
             // return Err(RoutingError::UnknownConnection);
         };
 
+        if client_check &&
+           self.check_valid_client_message(&pub_id, hop_msg.content.routing_message())
+               .is_err() {
+            return Err(RoutingError::InvalidStateForOperation);
```

Comment:
So previously to disconnected from a client sending invalid hop-msg (probably malicious client) - now you don't. Any reason for the change ?